tests/reftests/update-upgrade.test

```diff
@@ -506,3 +506,51 @@ The following actions will be performed:
 -> removed   ongoing.dev
 -> installed ongoing.dev
 Done.
+### <REPO/packages/d-foo/d-foo.6/opam>
+opam-version: "2.0"
+build: [ "false" ]
+### opam update
+[NOTE] ongoing.dev has previously been updated with --working-dir, not resetting unless explicitly selected
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[second] no changes from file://${BASEDIR}/REPO2
+[default] synchronised from file://${BASEDIR}/REPO
+
+<><> Synchronising development packages <><><><><><><><><><><><><><><><><><><><>
+[quux.dev5] synchronised (no changes)
+Now run 'opam upgrade' to apply any package updates.
+### opam upgrade
+The following actions will be performed:
+=== recompile 1 package
+  - recompile d-foo 6 [upstream or system changes]
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+[ERROR] The compilation of d-foo.6 failed at "false".
+
+
+
+
+<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
++- The following actions failed
+| - build d-foo 6
++- 
+- No changes have been performed
+# Return code 31 #
```

Comment:
To show more explicitly what the problem is:
```suggestion
# Return code 31 #
### opam upgrade
Nothing to do.
```

Reply:
If I'm guessing what the problem is wrongly, maybe there should be a comment in the test to tell what the cache file is about

Reply:
Ah, the overall PR diff doesn't convey it, indeed. The point is that `opam upgrade` should repeatedly produce error 31 - we're trying to recompile a package which is going to fail. Commit-by-commit:
- 3030619036aed9336c781476e8536d26067560d8 has the second `opam upgrade` claiming "Nothing to do"
- d6d4167d4f586074a23f6d5f913fc3bd3d330159 is what the fix should then do to it (second attempt to install also fails)
- 85cd847a8de491257812d0321d337f16c76b21f4 shows that if we delete the cache file, then we get the desired behaviour (i.e. the cache is the problem)

Reply:
Maybe a comment could be added that says something like "test to make sure we return exit code 31"

Reply:
A comment? A comment? Done 🙂

Reply:
Thanks!